lib/suspenders/adapters/heroku.rb

```diff
@@ -0,0 +1,67 @@
+module Suspenders
+  module Adapters
+    class Heroku
+      def initialize(app_builder)
+        @app_builder = app_builder
+      end
+
+      def set_heroku_remotes
+        remotes = <<-SHELL.strip_heredoc
+
+          # Set up the staging and production apps.
+          #{join_heroku_app('staging')}
+          #{join_heroku_app('production')}
+        SHELL
+
+        app_builder.append_file "bin/setup", remotes
+      end
+
+      def setup_heroku_specific_gems
+        app_builder.inject_into_file(
+          "Gemfile",
+          %{\n\s\sgem "rails_stdout_logging"},
+          after: /group :staging, :production do/,
+        )
+      end
+
+      def set_heroku_rails_secrets
+        %w(staging production).each do |environment|
+          run_heroku(
+            "config:add SECRET_KEY_BASE=#{generate_secret}",
+            environment,
+          )
+        end
+      end
+
+      private
+
+      attr_reader :app_builder
+
+      def join_heroku_app(environment)
+        heroku_app_name = heroku_app_name_for(environment)
+        <<-SHELL
+if heroku join --app #{heroku_app_name} &> /dev/null; then
+  git remote add #{environment} git@heroku.com:#{heroku_app_name}.git || true
+  printf 'You are a collaborator on the "#{heroku_app_name}" Heroku app\n'
+else
+  printf 'Ask for access to the "#{heroku_app_name}" Heroku app\n'
+fi
+        SHELL
+      end
+
+      def heroku_app_name_for(environment)
+        "#{app_builder.app_name.dasherize}-#{environment}"
+      end
+
+      def generate_secret
+        SecureRandom.hex(64)
+      end
+
+      def run_heroku(command, environment)
```

Comment:
Again, this came directly from the `app_builder`, but if we're changing it maybe `run_toolbelt_command` could work, since we already know it related to Heroku.